src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -220,51 +220,47 @@ static void fill_rc_input_msg(input_rc_s *rc, mavlink_rc_channels_t *rc_channels
 	rc->values[17] = rc_channels->chan18_raw;
 }
 
-void Simulator::update_sensors(const mavlink_hil_sensor_t *imu)
+void Simulator::update_sensors(const hrt_abstime &time, const mavlink_hil_sensor_t &imu)
 {
-	// write sensor data to memory so that drivers can copy data from there
-	RawMPUData mpu = {};
-	mpu.accel_x = imu->xacc;
-	mpu.accel_y = imu->yacc;
-	mpu.accel_z = imu->zacc;
-	mpu.temp = imu->temperature;
-	mpu.gyro_x = imu->xgyro;
-	mpu.gyro_y = imu->ygyro;
-	mpu.gyro_z = imu->zgyro;
-
-	write_MPU_data(&mpu);
-	perf_begin(_perf_mpu);
-
-	RawAccelData accel = {};
-	accel.x = imu->xacc;
-	accel.y = imu->yacc;
-	accel.z = imu->zacc;
-
-	write_accel_data(&accel);
-	perf_begin(_perf_accel);
-
-	RawMagData mag = {};
-	mag.x = imu->xmag;
-	mag.y = imu->ymag;
-	mag.z = imu->zmag;
+	if ((imu.fields_updated & 0x1FFF) != 0x1FFF) {
+		PX4_DEBUG("All sensor fields in mavlink HIL_SENSOR packet not updated.  Got %08x", imu.fields_updated);
+	}
 
-	write_mag_data(&mag);
-	perf_begin(_perf_mag);
+	// gyro
+	{
+		float scaling = 1000.0f;
```

Comment:
Why? And can be `const`.

Reply:
The driver helpers take raw int16 data, but mavlink HIL_SENSOR comes in already scaled as float. This is to prevent loss of precision and somewhat more like the actual drivers.